docs/sql-ref-syntax-ddl-alter-table.md

```diff
@@ -27,6 +27,9 @@ license: |
 
 `ALTER TABLE RENAME TO` statement changes the table name of an existing table in the database.
 
+If the table is cached, the commands clear cached data of the table and all its dependents that refer to it. The cache will be lazily filled when the next time the table or the dependents are accessed.
+The partition rename command keeps table dependents as cached but the table rename does not. In this way dependents should be re-cached explicitly after table renaming.
```

Comment:
I don't get it. Won't the cache be filled lazily? Why we need to re-cache explicitly?

Reply:
because the table rename command does not do that. It just cache table itself.

Reply:
Look at the implementation https://github.com/apache/spark/blob/2c243c93d9b107ed19fa21002d54deccd17ce50a/sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala#L201-L210

It uncaches tables (and its dependents), and caches ONLY the table back.

Reply:
hmm, shall we fix it?

Reply:
We cannot just replace the code above by `CatalogImpl.refreshTable()` or by `recacheByPlan` because the plan after renaming is different. I guess it includes `CatalogTable` which has different `identifier` after renaming.

Reply:
ah got it now.